mungegithub/mungers/approval-handler.go

```diff
@@ -64,15 +61,17 @@ func (*ApprovalHandler) AddFlags(cmd *cobra.Command, config *github.Config) {}
 
 // Munge is the workhorse the will actually make updates to the PR
 // The algorithm goes as:
-// - Initially, we set up approverSet
-//   - Go through all comments after latest commit. If any approver said "/approve", add him to approverSet.
-// - For each file, we see if any approver of this file is in approverSet.
+// - Initially, we build two sets, an approverSet and a cancelSet
+//   - Go through all comments after latest commit.
+//	- If anyone said "/approve", add them to approverSet.
+//	- If anyone said "/approve cancel", add them to cancelSet.
```

Comment:
So what happens if I do (in three different comments):
```
/approve
/approve cancel
/approve
```
First I'm in the `approverSet`, then I'm in `cancelSet`, and the last `/approve` is a no-op?

Suggested improvement:
- Add to `approverSet` when you see `/approve`.
- Remove from `approverSet` when you see `/approve cancel`
- Remove label if already applied and shouldn't be anymore